### PR TITLE
[Fluent 2 iOS] Updating BottomSheetController colors to Fluent 2 colors

### PR DESCRIPTION
**Pull request by edjamesmsft**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

Updated BottomSheetController to use the new Fluent 2 colors.

### Verification

Verified using Simulator and compared against design spec.

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Simulator Screen Shot - iPhone 13 Pro - 2022-07-15 at 13 57 07 Simulator Screen Shot - iPhone 13 Pro - 2022-07-15 at 13 56 57 | Simulator Screen Shot - iPhone 13 Pro - 2022-07-27 at 14 50 44 Simulator Screen Shot - iPhone 13 Pro - 2022-07-15 at 13 56 15 |

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [x] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [x] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1072)

### CODE REVIEW

File: ios/FluentUI/Bottom Sheet/BottomSheetController.swift [hunk 1]
```diff
@@ -873,6 +873,9 @@ public class BottomSheetController: UIViewController {
 
     private let shouldShowDimmingView: Bool
 
+    private lazy var shadowColor = UIColor(colorValue: self.view.fluentTheme.globalTokens.neutralColors[.black]).cgColor
```
Comment (harrieshin) [inline ios/FluentUI/Bottom Sheet/BottomSheetController.swift:876]: if we are going to define shadow colors this way, why do we even have shadow tokens? https://github.com/microsoft/fluentui-apple/blob/fluent2-colors/ios/FluentUI/Core/Theme/Tokens/AliasTokens.swift#:~:text=public-,enum,-ShadowColorsTokens%3A%20CaseIterable

Comment (mischreiber) [reply ios/FluentUI/Bottom Sheet/BottomSheetController.swift:876]: FWIW this isn't a tokenized control yet, right?

Comment (harrieshin) [reply ios/FluentUI/Bottom Sheet/BottomSheetController.swift:876]: talked offline with @mischreiber but even if it isn't tokenized control we should be using whatever set of defined shadow token color. If not, why even have those alias tokens.

**Review by laminesm — COMMENTED** [review]
Need to add observer for fluent theme change

File: ios/FluentUI/Bottom Sheet/BottomSheetController.swift [hunk 2]
```diff
@@ -409,10 +409,10 @@ public class BottomSheetController: UIViewController {
...
         // We need to have the shadow on a parent of the view that does the corner masking.
         // Otherwise the view will mask its own shadow.
-        bottomSheetView.layer.shadowColor = shadowColor
-        bottomSheetView.layer.shadowOffset = Constants.Shadow.offset
-        bottomSheetView.layer.shadowOpacity = Constants.Shadow.opacity
-        bottomSheetView.layer.shadowRadius = Constants.Shadow.radius
+        let shadow28 = view.fluentTheme.aliasTokens.shadow[.shadow28]
+        bottomSheetView.layer.shadowColor = UIColor(dynamicColor: shadow28.colorTwo).cgColor
+        bottomSheetView.layer.shadowOffset = CGSize(width: shadow28.xTwo, height: shadow28.yTwo)
+        bottomSheetView.layer.shadowRadius = shadow28.blurTwo
```
Comment (harrieshin) [inline ios/FluentUI/Bottom Sheet/BottomSheetController.swift:415]: can we add documentation in ShadowInfo structs? what does blur even mean? that means radius?

File: ios/FluentUI/Bottom Sheet/BottomSheetController.swift [hunk 3]
```diff
@@ -873,6 +873,8 @@ public class BottomSheetController: UIViewController {
 
     private let shouldShowDimmingView: Bool
 
+    private lazy var backgroundColor = UIColor(dynamicColor: self.view.fluentTheme.aliasTokens.colors[.background2])
```
Comment (harrieshin) [inline ios/FluentUI/Bottom Sheet/BottomSheetController.swift:876]: referencing self.view in the lazyvar seems weird to me

File: ios/FluentUI/Bottom Sheet/BottomSheetController.swift (hunk 2 above)
Comment (edjamesmsft) [reply ios/FluentUI/Bottom Sheet/BottomSheetController.swift:415]: Blur does mean total blur radius. I can add that to the ShadowInfo structs.

File: ios/FluentUI/Core/Theme/Tokens/ShadowInfo.swift [hunk 4]
```diff
@@ -6,6 +6,7 @@
 import CoreGraphics
 
 /// Represents a two-part shadow as used by FluentUI.
+/// Blur refers to the radius of the shadow's blur.
```
Comment (harrieshin) [inline ios/FluentUI/Core/Theme/Tokens/ShadowInfo.swift:9]: we should have documentation per variable in this entire struct

**Comment by harrieshin** [comment]
tested on iPad?

File: ios/FluentUI/Core/Theme/Tokens/ShadowInfo.swift [hunk 5]
```diff
@@ -7,12 +7,27 @@ import CoreGraphics
...
+    /// The size of the shadow on the y-axis (also, height) for shadow 1
     let yOne: CGFloat
+
+    /// The color of the shadow for shadow 2
     let colorTwo: DynamicColor
+
+    /// The radius of the shadow for shadow 2
     let blurTwo: CGFloat
+
+    /// The size of the shadow on the x-axis (also, width) for shadow 2
```
Comment (harrieshin) [inline ios/FluentUI/Core/Theme/Tokens/ShadowInfo.swift:28]: its the size? i thought it was offset?

**Comment by edjamesmsft** [comment]
Yes, this was tested on iPad (though only simulator), including rotating.

**Comment by harrieshin** [comment]
> Yes, this was tested on iPad (though only simulator), including rotating.

make sure you update the pr check list please!

File: ios/FluentUI/Core/Theme/Tokens/ShadowInfo.swift (hunk 5 above)
Comment (edjamesmsft) [reply ios/FluentUI/Core/Theme/Tokens/ShadowInfo.swift:28]: Fixed

**Review by harrieshin — APPROVED** [review]

**Pull request merged**